packages/storage/src/platform/node/connection.ts

```diff
@@ -0,0 +1,116 @@
+/**
+ * @license
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ErrorCode, XhrIo } from '../../implementation/xhrio';
+import { internalError } from '../../implementation/error';
+import * as nodeFetch from 'node-fetch';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const fetch: typeof window.fetch = nodeFetch as any;
+
+/**
+ * We use this instead of goog.net.XhrIo because goog.net.XhrIo is hyuuuuge and
+ * doesn't work in React Native on Android.
+ */
+export class FetchXhrIo implements XhrIo {
```

Comment:
Was wondering if we can use fetch for both browser and node. But guess not because fetch doesn't support progress reporting. :(